File: VB/WinSolution.Module.Win/OpenObjectFromPopupListViewController.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.ExpressApp.DC
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Actions
Imports DevExpress.Persistent.BaseImpl

Namespace WinSolution.Module.Win
	Partial Public Class OpenObjectFromPopupListViewController
		Inherits ViewController
		Public Sub New()
			InitializeComponent()
			RegisterActions(components)
			TargetViewNesting = Nesting.Root
			TargetViewType = ViewType.ListView
			TargetObjectType=GetType(DomainObject1)
		End Sub
		Private Sub popupWindowShowAction1_CustomizePopupWindowParams(ByVal sender As Object, ByVal e As CustomizePopupWindowParamsEventArgs) Handles paOpenPopupListView.CustomizePopupWindowParams
			Dim objectSpaceInternal As IObjectSpace = Application.CreateObjectSpace()
			Dim coll As CollectionSourceBase = New CollectionSource(objectSpaceInternal, GetType(Person))
			e.View = Application.CreateListView(Application.FindListViewId(GetType(Person)), coll, False)
		End Sub
	End Class
End Namespace

File: CS/WinSolution.Module.Win/OpenObjectFromPopupListViewController.cs
using DevExpress.ExpressApp.DC;
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Actions;
using DevExpress.Persistent.BaseImpl;

namespace WinSolution.Module.Win {
    public partial class OpenObjectFromPopupListViewController : ViewController {
        public OpenObjectFromPopupListViewController() {
            InitializeComponent();
            RegisterActions(components);
            TargetViewNesting = Nesting.Root;
            TargetViewType = ViewType.ListView;
            TargetObjectType=typeof(DomainObject1);
        }
        private void popupWindowShowAction1_CustomizePopupWindowParams(object sender, CustomizePopupWindowParamsEventArgs e) {
            IObjectSpace objectSpaceInternal = Application.CreateObjectSpace();
            CollectionSourceBase coll = new CollectionSource(objectSpaceInternal, typeof(Person));
            e.View = Application.CreateListView(
                Application.FindListViewId(typeof(Person)),
                coll,
                false
            );
        }
    }
}
